### PR TITLE
[SERVICES-992] MW helper gateway for registration

### PR DESCRIPTION
**Pull request by ArturKlajnerok**
Gateway to MW API methods that allow to check/update AntiSpoof extension and send confirmation email after successful registration.

/cc @Wikia/services-team

### CODE REVIEW

File: service/user-registration/src/main/java/com/wikia/services/userregistration/guice/UserRegistrationModule.java [hunk 1]
```diff
@@ -38,4 +42,24 @@ public int provideExpirationTime(UserRegistrationConfiguration configuration){
     return configuration.getExpirationTime();
   }
 
+  @Provides
+  @Named("wikia-helper-gateway")
```
Comment (nmonterroso) [inline service/user-registration/src/main/java/com/wikia/services/userregistration/guice/UserRegistrationModule.java]: Can we change this to be a constant?

File: service/user-registration/build.gradle [hunk 2]
```diff
@@ -9,13 +9,15 @@ applicationDefaultJvmArgs = [
 ]
 
 dependencies {
+    compile 'javax.json:javax.json-api:1.0'
+    compile 'org.glassfish:javax.json:1.0.4'
```
Comment (nmonterroso) [inline service/user-registration/build.gradle]: Is there a reason we're introducing a new library for json parsing? We could use jackson for this purpose to deserialize to simple beans using `ObjectMapper`

File: service/user-registration/src/main/resources/user-registration.yaml [hunk 3]
```diff
@@ -61,3 +71,5 @@ swaggerConfiguration:
     authKeyLocation: ${SWAGGER_SECURITY_TOKEN_TYPE:header}
 expirationTime: ${EXPIRATION_TIME:3600}
 cookieDomain: ${COOKIE_DOMAIN:"test.domain.com"}
+secretToken: ${SECRET_TOKEN:"token"}
+wikiaDomain: ${WIKIA_DOMAIN:"www.wikia.com"}
```
Comment (sqreek) [inline service/user-registration/src/main/resources/user-registration.yaml]: yes we should for dev-box testing

File: service/user-registration/build.gradle (hunk 2 above)
Comment (sqreek) [inline service/user-registration/build.gradle]: +1

File: service/user-registration/src/main/resources/user-registration.yaml (hunk 3 above)
Comment (ArturKlajnerok) [inline service/user-registration/src/main/resources/user-registration.yaml]: We can improve that with consul resolution as a separate subtask.  For now we can use consul/env to provide other configuration values.

File: service/user-registration/src/main/java/com/wikia/services/userregistration/guice/UserRegistrationModule.java (hunk 1 above)
Comment (ArturKlajnerok) [inline service/user-registration/src/main/java/com/wikia/services/userregistration/guice/UserRegistrationModule.java]: Fixed in : e767f8d and 0e22859

File: service/user-registration/src/main/java/com/wikia/services/userregistration/guice/UserRegistrationModule.java [hunk 4]
```diff
@@ -38,4 +37,23 @@ public int provideExpirationTime(UserRegistrationConfiguration configuration){
...
+  public WikiaHelperGateway provideWikiaHelperGateway(
+      UserRegistrationConfiguration configuration,
+      @Named("http-user-registration-service") CloseableHttpClient httpClient) {
+    return new WikiaHelperGateway(httpClient,
+                                  configuration.getWikiaDomain(),
+                                  configuration.getSecretToken());
+  }
+
+  @Provides
+  @Named("http-user-registration-service")
```
Comment (nmonterroso) [inline service/user-registration/src/main/java/com/wikia/services/userregistration/guice/UserRegistrationModule.java]: we should have constants for all the `@Named` dependencies

File: service/user-registration/build.gradle (hunk 2 above)
Comment (ArturKlajnerok) [inline service/user-registration/build.gradle]: Changed in : 485538f

File: service/user-registration/src/main/java/com/wikia/services/userregistration/guice/UserRegistrationModule.java (hunk 4 above)
Comment (ArturKlajnerok) [inline service/user-registration/src/main/java/com/wikia/services/userregistration/guice/UserRegistrationModule.java]: Fixed : 40dbb87

**Comment by nmonterroso** [comment]
Are timed out tests passing locally? if so lgtm

**Comment by ArturKlajnerok** [comment]
Test are timing out on Jenkins, but pass locally.

**Pull request merged**